angular-workspace/projects/ni/nimble-angular/table-column/anchor/nimble-table-column-anchor-navigation-guard.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import type { TableColumnAnchorCellView } from '@ni/nimble-components/dist/esm/table-column/anchor/cell-view';
+
+export type ClickDelegate = (rowRecordId: string | undefined) => boolean;
+
+/**
+ * Directive to allow client to intercept anchor clicks and do router navigation instead of
+ * letting the default action handle the navigation.
+ */
+@Directive({
+    selector: 'nimble-table-column-anchor[clickDelegate]'
+})
+export class NimbleTableColumnAnchorNavigationGuardDirective {
+    @Input()
+    public clickDelegate?: ClickDelegate;
+
+    @HostListener('delegated-event', ['$event.detail.originalEvent', '$event.detail.recordId'])
+    private onDelegatedEvent(delegatedEvent: Event, recordId: string | undefined): void {
+        if (delegatedEvent.type !== 'click') {
+            return;
+        }
+
+        const clickEvent = delegatedEvent as MouseEvent;
+        // Only call the clickDelegate for plain left clicks.
+        // Those are the only ones that should potentially use the router.
+        if (clickEvent.button !== 0 || clickEvent.ctrlKey || clickEvent.shiftKey || clickEvent.altKey || clickEvent.metaKey) {
+            return;
+        }
```

Comment:
guessing this logic is derived from here? https://github.com/angular/angular/blob/35a3fde5b71ef3b50282fe6f7b37ca1c92b8d2a0/packages/router/src/directives/router_link.ts#L453

If so can you put a pemallink referencing that